app/views/facility_account_users/new.html.haml

```diff
@@ -26,7 +26,7 @@
   = f.input :user_id, as: :hidden, input_html: { value: @user.id, name: "user_id" }
 
   = f.input :user_role,
-    collection: AccountUser.selectable_user_roles(current_user, current_facility),
+    collection: AccountUser.selectable_user_roles(current_user, current_facility).map {|role| [text("accounts.member_table.role.#{role.downcase.parameterize.underscore}"), role]},
```

Comment:
Yeah, this isn't great. What would you think about using a presenter?

```ruby
class AccountUserPresenter < SimpleDelegator
  def self.localized_role(role)
    I18n.t(role.downcase.parameterize.underscore, scope: "accounts.role") # Move the en.yml location to be more generic than `member_table`
  end

  def self.selectable_user_roles(granting_user = nil, facility = nil)
    # Nothing other than the collection uses `selectable_user_roles`, so it could even be pulled up into this class
    AccountUser.selectable_user_roles(granting_user, facility).map {  { |role| [localized_role(role), role] }
  end

  def localized_role
    self.class.localize_role(user_role)
  end
end
```

In the `member_table`:

`= AccountUserPresenter.new(au).localized_role`

In the form:
`AccountUserPresenter.selectable_user_roles(current_user, current_facility)`

There's also `account_users/new` that uses `selectable_user_roles`. That's for account owners to manage access to their own accounts.

* Note: I haven't actually tested that code

Reply:
Wow, besides one semicolon this code worked perfect! Look good now?